### ISSUE TITLE
Adding confidence check to test loop

### ISSUE THREAD

**Issue by AWikramanayake** [AWikramanayake/CNN-Visualisations#3]
Implement a guess confidence check from logits (similar to that in get_pretrained_guesses) into the test step for MNIST/FashionMNIST datamodule and add a hook to perform visualisations on the weakest guesses
